packages/@core.keystone/tasks.js

```diff
@@ -0,0 +1,137 @@
+const Queue = require('bull')
+
+const conf = require('@core/config')
+
+const TASK_TYPE = 'TASK'
+const WORKER_CONCURRENCY = parseInt(conf.WORKER_CONCURRENCY || '2')
+// NOTE: If this is True, all tasks will be executed locally by blocking until the task returns.
+// Tasks will be executed locally instead of being sent to the queue.
+const FAKE_WORKER_MODE = conf.FAKE_WORKER_MODE
+const WORKER_REDIS_URL = conf.WORKER_REDIS_URL
+if (!WORKER_REDIS_URL) throw new Error('No WORKER_REDIS_URL environment')
+const taskQueue = new Queue('tasks', WORKER_REDIS_URL)
+const globalTaskOptions = { attempts: 3, backoff: true }
+const TASKS = new Map()
+
+function createTask (name, fn, opts = {}) {
```

Comment:
Please, use class to group same logic pieces which cannot work each without other.

Reply:
see: https://github.com/open-condo-software/condo/blob/master/apps/notification_bot/botController.ts